src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/CollidingNodeSystemAlgorithm/Vertex.java

```diff
@@ -0,0 +1,58 @@
+package edu.greenblitz.tobyDetermined.Nodesssss.CollidingNodeSystemAlgorithm;
+
+import static edu.greenblitz.tobyDetermined.Nodesssss.NodeSystemUtils.*;
+import static edu.greenblitz.tobyDetermined.Nodesssss.NodeBase.SystemsPosition;
+
+
+import java.util.LinkedList;
+
+public class Vertex {
+    private final SystemsPosition startPosition;
+    private final SystemsPosition endPosition;
+    private SystemsPosition system2Position;
+
+    public Vertex(SystemsPosition startPosition, SystemsPosition endPosition, SystemsPosition system2Position) {
+        this.startPosition = startPosition;
+        this.endPosition = endPosition;
+        this.system2Position = system2Position;
+    }
+
+    public SystemsPosition getStartPosition() {
+        return startPosition;
+    }
+
+    public SystemsPosition getEndPosition() {
+        return endPosition;
+    }
+
+    public double getTimeCost() {
+        return getCostByMap(startPosition, endPosition);
+    }
+
+    public SystemsPosition getSystem2Position() {
+        return system2Position;
+    }
+
+    public void setSystem2Position(SystemsPosition otherSystem) {
+        this.system2Position = otherSystem;
+    }
+
+    public boolean isPositionFineForVertex(SystemsPosition position) {
```

Comment:
"Fine" is not that clear, how about 'isValidForVertex' or maybe even just 'isValid'